kotlin/codegen/src/main/java/com/squareup/moshi/kotlin/codegen/api/AdapterGenerator.kt

```diff
@@ -158,8 +159,13 @@ internal class AdapterGenerator(
     return FunSpec.builder("toString")
         .addModifiers(KModifier.OVERRIDE)
         .returns(String::class)
-        .addStatement("return %S",
-            "GeneratedJsonAdapter(${originalTypeName.rawType().simpleNames.joinToString(".")})")
+        .addStatement(
+            "return %M·{ append(%S).append(%S).append(%S) }",
```

Comment:
Can also pass the exact size as the first argument to `buildString` to avoid expansion / over-allocation.